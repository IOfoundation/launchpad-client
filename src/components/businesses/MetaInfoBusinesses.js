import React from 'react';
import {PropTypes} from 'prop-types';

class MetaInfoBusinesses extends React.Component {
  resourcesStartOn() {
    const _businessesMetadata = this.props.businessesMetadata;
    return (
      (_businessesMetadata.pagination.current_page - 1) *
      (_businessesMetadata.pagination.per_page + 1)
    );
  }

  resourcesEndOn() {
    const _businessesMetadata = this.props.businessesMetadata;
    let resourcesEndOn =
      _businessesMetadata.pagination.current_page *
      _businessesMetadata.pagination.per_page;
    const totalResources = _businessesMetadata.totalOrganizations;
    if (totalResources < resourcesEndOn) {
      resourcesEndOn = totalResources;
    }
    return resourcesEndOn;
  }

  render() {
    return (
      <span className="secondary">{`Showing Results ${this.resourcesStartOn()}-${this.resourcesEndOn()}`}</span>
    );
  }
}

MetaInfoBusinesses.propTypes = {
  businessesMetadata: PropTypes.object.isRequired,
};

export default MetaInfoBusinesses;
