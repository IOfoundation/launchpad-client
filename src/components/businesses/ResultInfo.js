import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import {FaSortDesc} from 'react-icons/lib/fa';

class ResultInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showOptions: false,
    };
  }
  _toggleOptions() {
    this.setState({showOptions: !this.state.showOptions});
  }
  _renderOptions() {
    return (
      <div className="filterSelect_dropdown">
        <p className="smallFont primary filterSelect_dropdown_title">
          {'Sort Results'}
        </p>
        <button className="filterSelect_option">
          <span className="filterSelect_text">{'Test filter'}</span>
        </button>
      </div>
    );
  }
  render() {
    const {businessesMetadata} = this.props;
    return (
      <div className="businessList_top">
        <div className="row between-xs middle-xs businessList_top_content col-xs-6">
          <span className="secondary bold bodyFont">{`${businessesMetadata.total_organizations} Resources Available`}</span>
          <div
            className="businessList_top_filter row middle-xs noPadding"
            onClick={() => this._toggleOptions()}
          >
            <p className="businessList_top_filter_text">
              {'Sort Results'}
            </p>
            <FaSortDesc
              className="businessList_top_filter_icon"
              size={16}
              color={'#000'}
            />

            {this.state.showOptions ? this._renderOptions() : null}
          </div>
        </div>
      </div>
    );
  }
}

ResultInfo.propTypes = {
  businessesMetadata: PropTypes.object.isRequired,
};

export default ResultInfo;
