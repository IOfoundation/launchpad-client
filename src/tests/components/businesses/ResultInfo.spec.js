import React from 'react';
import {shallow} from 'enzyme';

import ResultInfo from '../../../components/businesses/ResultInfo';
import MetaInfoBusinesses
  from '../../../components/businesses/MetaInfoBusinesses';

describe('<ResultInfo />', () => {
  it('Renders MetaInfoBusinesses component', () => {
    const wrapper = shallow(
      <ResultInfo businessesMetadata={{pagination: {current_page: 1}}} />
    );
    expect(wrapper.find(MetaInfoBusinesses).length).toEqual(1);
  });

  it('Display the total Businesses found', () => {
    const wrapper = shallow(
      <ResultInfo
        businessesMetadata={{
          total_organizations: 10,
          pagination: {current_page: 1},
        }}
      />
    );
    expect(wrapper.contains('10 Resources Available')).toBe(true);
  });
});
