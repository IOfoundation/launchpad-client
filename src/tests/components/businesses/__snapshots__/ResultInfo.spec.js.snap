// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultInfo /> renders snapshot of ResultInfo 1`] = `
<div
  className="grid business-type-btn"
>
  <p
    className="m-bot-16 col-xs-12 col-md-12 col-lg-12 p-0 business-type-label"
  >
    Select a business type that represents you.
  </p>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={4}
      xs={12}
    >
      <button
        className="btn btn-search btn-outline btn--padding"
        key="142"
        onClick={[Function]}
      >
        <div
          className="btn-search-text text-bold"
        >
          <span
            className="desktop-devices"
          >
            Startup or High
          </span>
          <span
            className="desktop-devices"
          >
            Growth Business
          </span>
          <span
            className="mobile-devices"
          >
            Startup or High Growth Business
          </span>
        </div>
        <ArrowRight
          className="btn-search-icon"
          size={20}
          style={
            Object {
              "color": "#000",
              "verticalAlign": "middle",
            }
          }
        />
      </button>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={4}
      xs={12}
    >
      <button
        className="btn btn-search btn-outline btn--padding"
        key="143"
        onClick={[Function]}
      >
        <div
          className="btn-search-text text-bold"
        >
          <span
            className="desktop-devices"
          >
            Main Street/
          </span>
          <span
            className="desktop-devices"
          >
            Small Business
          </span>
          <span
            className="mobile-devices"
          >
            Main Street/Small Business
          </span>
        </div>
        <ArrowRight
          className="btn-search-icon"
          size={20}
          style={
            Object {
              "color": "#000",
              "verticalAlign": "middle",
            }
          }
        />
      </button>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={4}
      xs={12}
    >
      <button
        className="btn btn-search btn-outline btn--padding"
        key="144"
        onClick={[Function]}
      >
        <div
          className="btn-search-text text-bold"
        >
          <span
            className="desktop-devices"
          >
            Microenterprise or
          </span>
          <span
            className="desktop-devices"
          >
            Home-Based Business
          </span>
          <span
            className="mobile-devices"
          >
            Microenterprise or Home-Based Business
          </span>
        </div>
        <ArrowRight
          className="btn-search-icon"
          size={20}
          style={
            Object {
              "color": "#000",
              "verticalAlign": "middle",
            }
          }
        />
      </button>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
